src/reducers/composers.js

```diff
@@ -86,32 +77,134 @@ export const addDataToMapComposed = (state, action) => {
 
   const {datasets, options, config} = action.payload;
   let parsedConfig = config;
-
   if (config && config.config && config.version) {
-    // if passed in saved config
-    parsedConfig = KeplerGlSchema.parseSavedConfig(config)
+// if passed in saved config
+    parsedConfig = KeplerGlSchema.parseSavedConfig(config);
   }
-  // Update visState store
+// Update visState store
   let mergedState = updateVisDataComposed(state, {datasets, options, config: parsedConfig && parsedConfig.visState});
-
   // Update mapState store
   mergedState = {
     ...mergedState,
     mapState: stateMapConfigUpdater(mergedState.mapState, {payload: {mapState: parsedConfig && parsedConfig.mapState}})
   };
-
   // Update mapStyle store
   mergedState = {
     ...mergedState,
     mapStyle: styleMapConfigUpdater(mergedState.mapStyle, {payload: {mapStyle: parsedConfig && parsedConfig.mapStyle}})
   };
+  return mergedState;
+};
+/**
+ * Remove layer data from layer data
+ * @param state
+ * @param action {datasets, options, config}
+ * @returns state
+ */
+export const removeLayerDataComposed = (state, action) => {
+  // extract mapData
```

Comment:
I don't think this is a reasonable way to remove rows from a dataset and update related layers. For one thing, you only removed rows from `layerData` which should not be directly edited, layerData should only be updated by `visState` reducer based on current filtered dataset and layer config. If you didn't remove the actual dataset rows that the layer is associated with. upon any filter or layer config changes, the layer will display the original dataset.

A reasonable way to do this, similar to `addRowsToDataset` is to edit the existing dataset, and call `addDataToMapComposed` to let the visState reducer handles the rest. 

You should also rename this function to `removeRowsFromDataset`. In terms of how to remove date, ideally not by direct comparison and `arrayEquals`, VERY SLOW. 

The argument can be 1. an array of `index` to filter `data.rows`, or 2. a `fieldName` and `uniqueValues` to filter the field value based on.  Option 1 requires the user to keep track of the data added to kepler.gl. Option 2 requires the existing dataset contains a field that has unique values. e.g. `id`, `name`, `index` etc. Either way is fine, rather based on your use cases

Reply:
Thanks @heshan0131 I totally missed filters case!, Initially I tried removing from `datasets` since I could not find a way to rerender the graph ended up in removing from `layerData`. 
Thanks for your detailed review now I am thinking to remove rows based on lat longs/geojson instead of comparing entire array since either of the ones(latlong/geojson) will be present for every Kepler graph, do you see any pitfalls with this approach?